sdk/common/src/test/scala/org/typelevel/otel4s/sdk/ArbitraryInstances.scala

```diff
@@ -90,4 +94,45 @@ trait ArbitraryInstances {
     schemaUrl <- Gen.option(nonEmptyString)
   } yield Resource(attrs, schemaUrl))
 
+  implicit val attributeTypeCogen: Cogen[AttributeType[_]] =
+    Cogen[String].contramap(_.toString)
+
+  implicit def attributeKeyCogen[A]: Cogen[AttributeKey[A]] =
+    Cogen[(String, String)].contramap[AttributeKey[A]] { attribute =>
+      (attribute.name, attribute.`type`.toString)
+    }
+
+  implicit def attributeCogen[A: Cogen]: Cogen[Attribute[A]] =
+    Cogen[(AttributeKey[A], A)].contramap(a => (a.key, a.value))
+
+  implicit val attributeExistentialCogen: Cogen[Attribute[_]] =
+    Cogen { (seed, attr) =>
+      def primitive[A: Cogen](seed: Seed): Seed =
+        Cogen[A].perturb(seed, attr.value.asInstanceOf[A])
+
+      def list[A: Cogen](seed: Seed): Seed =
+        Cogen[List[A]].perturb(seed, attr.value.asInstanceOf[List[A]])
+
+      val valueCogen: Seed => Seed = attr.key.`type` match {
+        case AttributeType.Boolean     => primitive[Boolean]
+        case AttributeType.Double      => primitive[Double]
+        case AttributeType.String      => primitive[String]
+        case AttributeType.Long        => primitive[Long]
+        case AttributeType.BooleanList => list[Boolean]
+        case AttributeType.DoubleList  => list[Double]
+        case AttributeType.StringList  => list[String]
+        case AttributeType.LongList    => list[Long]
+      }
+
+      valueCogen(attributeKeyCogen.perturb(seed, attr.key))
+    }
+
```

Comment:
@armanbilge I didn't use `Cogen` before. Am I defining them correctly? 

Do I understand this correctly:
1) `Gen` - produces a value of type `A` from `Seed`
2) `Cogen` - produces a `Seed` for the given value `A`

Is `Cogen` used to adjust the seed for the structured values, and, as a result, produce better pseudo-random values?

Reply:
> Is `Cogen` used to adjust the seed for the structured values, and, as a result, produce better pseudo-random values?

`Cogen` is used to generate arbitrary lambdas e.g. `Arbitrary[A => B]`. If you have `Cogen[A]` and `Gen[B]` then you can chain `A => Seed`, `Seed` => `Seed`, and `Seed => B` to get an arbitrary but deterministic `A => B` function.

---

I think your implementation looks good, typically I try to avoid the manual plumbing of seeds and instead transform to some datatype for which `Cogen` already exists. In this case that would be very messy 😁 but it could look something like this:
```scala
Either[
  Boolean,
  Either[
    Double,
    Either[
      String,
      Either[
        Long,
        Either[...]
      ]
    ]
  ]
]
```